src/components/questions/ImageUpload.tsx

```diff
@@ -0,0 +1,119 @@
+import React from 'react';
+import {
+  StyleSheet,
+  Text,
+  View,
+  TouchableOpacity,
+  Image,
+  ToastAndroid,
+  Alert,
+  PermissionsAndroid,
+} from 'react-native';
+import { launchCamera, launchImageLibrary } from 'react-native-image-picker';
+import Icon from 'react-native-vector-icons/FontAwesome';
+import COLORS from '../../assets/colors';
+
+const ImageUpload = () => {
+  const [image, SetImage] = React.useState('');
+
+  // Show toast alert messages
+  const setToastMsg = (msg: any) => {
+    ToastAndroid.showWithGravity(msg, ToastAndroid.SHORT, ToastAndroid.CENTER);
+  };
+
+  const uploadImage = () => {
+    launchImageLibrary({ mediaType: 'photo', quality: 1, includeBase64: true }, (response) => {
+      if (response.didCancel) {
+        setToastMsg('geen foto gekozen');
+      } else if (response.errorCode == 'permission') {
+        setToastMsg('permissie afgewezen');
+      } else if (response.errorCode == 'others') {
+        setToastMsg(response.errorMessage);
+      } else if (response.assets[0].fileSize > 2097152) {
```

Comment:
Weet je zeker dat de assets array bestaat? Krijg typescript errors dus blijkbaar kan t ook undefined zijn

Reply:
dat is nu aangepast naar of string of undefined in de useState. omdat het inderdaad undefined kan zijn als het niet bestaat.